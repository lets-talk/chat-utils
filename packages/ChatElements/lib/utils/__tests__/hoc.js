import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  withInfiniteScroll,
  withScrollWatch,
  withLoading,
  withPaginated,
  withAutoScroll,
} from '../hoc';

describe('utils/hoc', () => {
  const MySimpleComponent = () => <div>Super Simple</div>;
  const autoScrollOptions = { threshold: 300, direction: 'bottom' };
  const WithScroll = withAutoScroll(autoScrollOptions)(MySimpleComponent);

  it('withAutoScroll: should adds autoscroll feature to a wrapper component', () => {
    const mockCallBack = jest.fn();
    const fakeEventObject = {
      scrollHeight: 100, scrollTop: 10, offsetHeight: 10, preventDefault() {}, stopPropagation() {},
    };

    const wrapper = shallow(<WithScroll />);
    wrapper.simulate('scroll', fakeEventObject);
    expect(mockCallBack.mock.calls.length).toEqual(1);

    expect(window.scrollTo).toHaveBeenCalledWith(0, 0);
  });
});
