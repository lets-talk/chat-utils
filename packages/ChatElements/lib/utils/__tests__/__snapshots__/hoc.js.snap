// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/hoc withAutoScroll: should adds autoscroll feature to a wrapper component 1`] = `
<MySimpleComponent
  cmpRef={[Function]}
/>
`;

exports[`utils/hoc withLoading: should adds loading feature to a wrapper component 1`] = `
<div>
  <MySimpleComponent
    loading={true}
  />
  <div
    className="interactions"
  />
</div>
`;

exports[`utils/hoc withPaginated: should adds loading feature to a wrapper component 1`] = `
<div>
  <MySimpleComponent />
  <div
    className="interactions"
  >
    <button
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`utils/hoc withScrollWatch: should adds loading feature to a wrapper component 1`] = `
<MySimpleComponent
  x={0}
  y={0}
/>
`;

exports[`utils/hoc withScrollWatch: should adds loading feature to a wrapper component 2`] = `<MySimpleComponent />`;
